### ISSUE TITLE
[Request] Please assist in, or add the option to, try to base on top of Firefox Nightly

### ISSUE THREAD

**Issue by IronOverheat** [CachyOS/CachyOS-Browser-Common#1]
There might be use cases where Firefox Nightly excels higher than stable in performance, security and privacy, as the newest freshest code lands there first, and stability is often reasonable and crashing is infrequent (I say from multiple years of using Nightly). 

An example of fresh code that empowers privacy is the new Cookie Banner Reduction which could function as a nice fallback to automatically refuse cookie banners that are either manually allowed to unblock a website or that escape uBlock's Annoyances filter.

Please consider any of the following in order of effort to implement:

- Simply giving any advice/'dumbcumentation' on how to (dirtily) alter cachybrowser PKGBUILD to target the Nightly source code
- Adding "cachybrowser-git" to AUR ('dirty' port of the regular Cachy Browser PKGBUILD)
- || but as a PKGBUILD accessible through the cachyos repos if interested in 'adopting' the experimental 'you try it' PKGBUILD
- Go the extra mile and involve yourselves in any meaningful capacity if the opportunity and maintainer time/funds arrive for investigating this, such as offering a prebuilt cachybrowser-nightly

Ideally I'd do this myself as Ive done my fair share of PKGBUILD 'hacking' but I felt a smidge intimidated by this PKGBUILD and the fact I have no idea if I could reasonably compile Firefox everyday.

Thank you for reading and thank you kindly CachyOS devs for the awesome OS and apps.

**Comment by IronOverheat** [comment]
As a long-time privacy enthusiast and tweaker I also have many suggestions and ideas on how to improve CachyBrowser and a few things about the OS, unfortunately I am not a dev, but I'm quite used to editing and understanding simple code like in PKGBUILDs and other non-complex syntax. I am also free full-time and willing to volunteer with CachyOS if I meet the criteria, specially if I can receive pointers on how to actually become a dev/programmer (I am specially interested in cutting/bleeding edge memory safe langs). (Apologies for offtopic)

**Comment by vnepogodin** [comment]
Yes. I agree sometimes nightly can be better, new features. But also maintaining patchset requires some time and effort to patchset frequently, also because nightly does change code every day.

Although I was maintaining librewolf-hg, maintaining librewolf patchset to nightly

Also I don't "understand" why should we create the PKGBUILD that will try to patch and continue?

Best regards

**Comment by IronOverheat** [comment]
Thank you for the quick response, I'm terribly sorry I didn't reply earlier!

About the PKGBUILD, what I was referring to was simply copy+pasting the existing cachybrowser PKGBUILD found on your CachyOS apps github repo but since you would already be familiar with how you organized the PKGBUILD script, it would be easier for you to just make it very 'stupidly' just target the Nightly base without modifications to the actual patchset (which would obviously break the patchset in the apply step, and that's where users can "try it themselves" by attempting to 'fix' the patchset to apply properly by running makepkg and analyzing each error one by one).

I could just sit down for an afternoon and attempt to replicate the build environment and understand the script style from the native PKGBUILD (by comparing it to other Firefox-based PKGBUILDS, as well), but I figured per chance you may be interested in giving just a general 'go ahead' by making an easy port of the cachyos PKGBUILD and leave the compatibility troubleshooting with the users (which would include me, and I'd be happy to publish a PKGBUILD that successfully applies the patchset on top of Nightly)

**Comment by mauro-balades** [comment]
Nightly is updated almost daily, having to keep it up to date and test could be a very dubious job and they'll have to update 2 browsers now. There's no such thing as building it 'stupidly', because the patches applied might fail, some firefox API may change or just the UI in general.
